compliance/notify.py

```diff
@@ -1018,30 +1018,44 @@ def _get_alerts(self, accreditation):
         conf = self._config[accreditation]
         pd_service = conf['service_id'] if isinstance(conf, dict) else conf
 
-        # Get all triggered alerts
-        # NOTE: statuses[]=triggered is supposed to work, but doesn't
-        # so we have to filter the response. leaving it in the query,
-        # as it will be a bit more efficient if they fix the issue.
-        pages = pagerduty.get(
-            'alerts',
+        # Get all current incidents for the service, we need to specify both
+        # acknowledged and triggered so we do not send a second page if there
+        # has already been one acknowledged
+        incidents_data = pagerduty.get(
+            'incidents',
             params={
                 'service_ids[]': pd_service,
-                'statuses[]': 'triggered',
+                'statuses[]': ['acknowledged', 'triggered'],
                 'time_zone': 'UTC'
             },
             creds=self._creds
         )
-        alerts = [
-            {
-                'test': alert['alert_key'],
-                'details': alert.get('body', {}).get('details', ''),
-                'created_at': alert['created_at']
-            }
-            for page in pages
-            for alert in page.json().get('alerts', [])
-            if alert['status'] == 'triggered'
-        ]
-
+        alerts = []
+
+        # If there are any incidents then loop through them and fetch all the
+        # alerts that are related to the passed accreditation
+        for inc_data in incidents_data:
+            incidents = inc_data.json()['incidents']
+            for inc in incidents:
+                alerts_data = pagerduty.get(
+                    f'incidents/{inc["id"]}/alerts',
+                    params={
+                        'service_ids[]': pd_service,
+                        'statuses[]': ['acknowledged', 'triggered'],
+                        'time_zone': 'UTC'
+                    },
+                    creds=self._creds
+                )
+                # Only fetch alerts if it matches the accreditation
+                for alert_data in alerts_data:
+                    alerts += [
+                        {
+                            'test': a['alert_key'],
+                            'details': a.get('body', {}).get('details', ''),
+                            'created_at': a['created_at']
+                        } for a in alert_data.json()['alerts'] if a['body']
+                        ['cef_details']['source_origin'] == accreditation
+                    ]
```

Comment:
In general, I'd rather to keep the strategy of having nested list comprehension here in two separated `for` loops. First collect all incidents, then collect all tasks.

Reply:
Also, I'm not 100% sure but once the linter and formatters run they may have issues with the wrapping of the `if` condition across two lines.  So another reason to switch to separated for loops.

Reply:
I guess the linter/formatter don't care...

But I agree I'd that using separate loops would make this more readable.

Reply:
It was actually the formatter that changed it to be wrapped. I can update it to be two loops.

Reply:
Yeah that'll happen from time to time with the formatter.  It's useful 90% of the time but the rest of the time it comes up with sub-optimal (IMO) updates.

Reply:
This if statement is not even needed and would just be bloat as the main `notify` command already looks at what test the alert is for. Will remove my added if statement for origin all together.